src/dialogs/permissions.jsx

```diff
@@ -99,16 +121,57 @@ const EditPermissionsModal = ({ dialogResult, selected, path }) => {
         }
     };
 
-    function permissions_options() {
-        return [
-            ...map_permissions((value, label) => (
+    function permissions_options(mode) {
+        const options = [
+            <FormSelectOption
+              key="read-write"
+              value="read-write"
+              label={_("Read and write")}
+            />,
+            <FormSelectOption
+              key="read-only"
+              value="read-only"
+              label={_("Read-only")}
+            />,
+            <FormSelectOption
+              key="no-access"
+              value="no-access"
+              label={_("No access")}
+            />
+        ];
+
+        // Show write-only when such a file exists, but never offer this as a default option.
+        if (mode === 2 || mode === 3) {
+            options.push(
                 <FormSelectOption
-                  key={value}
-                  value={value}
-                  label={label}
+                  key="write-only"
+                  value="write-only"
+                  label={_("Write-only")}
                 />
-            ))
-        ];
+            );
+        }
+
+        return options;
+    }
+
+    function setPermissions(mask, shift, option) {
```

Comment:
This function discards existing execute bits if the file doesn't have `+x` for all three (owner, group, other). It's probably a side effect of how the new design decides if file is executable or not. Not a blocker for me but I wanted to point it out.

Eg: when trying to edit 750 file to 770 using the UI you'll end up with 760 file instead. Unexpected but probably makes sense with how the "set executable" checkbox works.